### PR TITLE
[#2002] Tracevent ghosting/overlapping should trigger properly now when ...

### PR DESCRIPTION
**Pull request by dseif**
...moving a trackevent between tracks

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/6754b1b85f4366a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/b160d89e7509914/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/6754b1b85f4366a/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/b160d89e7509914/output.txt

Total script time: 3.00 mins

File: src/timeline/trackevent-drag-manager.js [hunk 1]
```diff
@@ -17,11 +17,14 @@ define( [ "./ghost-track" ], function( GhostTrack ) {
 
       if ( trackView.findOverlappingTrackEvent( trackEventView ) ) {
         track.removeTrackEvent( trackEvent );
+        nextTrack = _media.getNextTrack( track ) || _media.addTrack();
+        trackView = nextTrack.view;
         while ( trackView.findOverlappingTrackEvent( trackEventView, true ) ) {
-          nextTrack = _media.getNextTrack( nextTrack ) || _media.addTrack();
+          nextTrack = _media.getNextTrack( track ) || _media.addTrack();
```
Comment (secretrobotron) [inline src/timeline/trackevent-drag-manager.js:23]: I'm not sure this is correct. It looks like this `while` loop only happens once or twice now, and moves on, calling `correctOverlapingTrackEvents` recursively. The `while` loop was meant to circumvent recursion. `nextTrack` should be updated and used in `_media.getNextTrack` to evaluate what happens after one instance of the loop occurs.

Am I reading this right?

**Pull request closed without merging**